x/dualstaking/keeper/delegator_reward.go

```diff
@@ -157,7 +158,7 @@ func (k Keeper) CalcProviderRewardWithDelegations(ctx sdk.Context, providerAddr
 	}
 	stakeEntry, err := k.epochstorageKeeper.GetStakeEntryForProviderEpoch(ctx, chainID, providerAddr, epoch)
 	if err != nil {
-		return math.ZeroInt(), utils.LavaFormatError(types.ErrCalculatingProviderReward.Error(), err,
```

Comment:
why we dont return the underlying error?

Reply:
because that error is not of type ErrProviderNotStaked (the caller checks the error type and acts differently for this error). I can add the underlying error as an attribute, is this ok?

Reply:
than change the underlying function to return the appropriate error

Reply:
done. see https://github.com/lavanet/lava/commit/fc789820494c93a334c568ac4e7a19d36e85a9a7

Reply:
also this: https://github.com/lavanet/lava/commit/1b44d97c2a97628c2e18943e6b04f6ee0ec0ca9b